core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java

```diff
@@ -55,7 +55,7 @@
 
     private int maxHashPartitionCount = 100;
     private int minHashPartitionCount = 4;
-    private Duration minQueryExpireAge = new Duration(15, TimeUnit.MINUTES);
+    private Duration minQueryExpireAge = new Duration(3, TimeUnit.MINUTES);
```

Comment:
What's the motivation for changing the default value here? In any case, this change should be in a separate commit and mentioned in the release notes.

Reply:
Information about query (a rather big objects like QueryInfo, QueryStats) is held by 15 minutes without looking at `maxQueryHistory` into account, which makes a huge difference for query purger effectiveness of it's loop which constantly tries to remove old queries or marked them as consumed in `ExecutingStatementResource`.

Reply:
This is a pretty visible change to users in any case. The problem with memory pressure from query history is one thing, but the synchronization problem shouldn't really change based on this value. The cases where `Query` becomes contended are when the query itself is still alive and results are being pulled, which means `resultsConsumed` will be false and the query purger thread will continue to keep contending with threads trying to fetch results for the `Query` monitor.

Anyways, in the case where `resultsConsumed == true` only the query purger thread is likely to be be attempting to acquire the lock in the first place, so no contention issue there, but also nothing really resolved either. Maybe the better option to reduce contention is to either:
- reduce the frequency of checking for `Query#markResultsConsumedIfReady()` in query purger to something larger than 200ms
- add a lock-free check path to only synchronize in `Query#markResultsConsumedIfReady()` if no results have been fetched within some time interval so that the purger thread can avoid locking on instances that are still having query results pulled from them regularly

Reply:
Reverted change, if this will be beneficial I will present separate PR for it.

Reply:
i remember Dain saying that this has something to do with client protocol. Perhaps, for example instantaneous expiration could be a problem if client retries a request.